### ISSUE TITLE
Variables used in string formatting via locals() are not detected

### ISSUE THREAD

**Issue by ozmartian** [PyCQA/pylint#1648]
Formatting strings with local var references through locals() are not detected so are reported as unused.

### Steps to reproduce
Simple example code writeup:

```
var1 = "using"
var2 = "local"
"a string {var1} a few {var2} vars".format(**locals())
```

### Current behavior
Variables reported as unused

### Expected behavior
Variables recognised as used and not reported

### pylint --version output
pylint 1.7.2,
astroid 1.5.3
Python 3.6.2 (default, Jul 20 2017, 03:52:27)
[GCC 7.1.1 20170630]

**Comment by PCManticore** [comment]
Yes, this is intended. We don't currently support that and it would be a hassle to make it work.

**Comment by ozmartian** [comment]
thats cool... hassle always trumps a simple ignore :-) thanks for helping make pylint what it is!
